xonsh/jobs.py

```diff
@@ -1,215 +1,13 @@
-        job['bg'] = True
-        job['status'] = 'stopped'
-        print()  # get a newline because ^Z will have been printed
-        print_one_job(act)
-    elif os.WIFSIGNALED(s):
-        print()  # get a newline because ^C will have been printed
-    if obj.poll() is not None:
-        builtins.__xonsh_active_job__ = None
-    _give_terminal_to(_shell_pgrp)  # give terminal back to the shell
-
-
-def kill_all_jobs():
-    """
-    Send SIGKILL to all child processes (called when exiting xonsh).
-    """
-    _clear_dead_jobs()
-    for job in builtins.__xonsh_all_jobs__.values():
-        os.kill(job['obj'].pid, signal.SIGKILL)
-
-
-def jobs(args, stdin=None):
-    """
-    xonsh command: jobs
-
-    Display a list of all current jobs.
-    """
-    _clear_dead_jobs()
-    for j in sorted(builtins.__xonsh_all_jobs__):
-        print_one_job(j)
-    return None, None
-
-
-def fg(args, stdin=None):
-    """
-    xonsh command: fg
-
-    Bring the currently active job to the foreground, or, if a single number is
-    given as an argument, bring that job to the foreground.
-    """
-    _clear_dead_jobs()
-    if len(args) == 0:
-        # start active job in foreground
-        act = builtins.__xonsh_active_job__
-        if act is None:
-            return '', 'Cannot bring nonexistent job to foreground.\n'
-    elif len(args) == 1:
-        try:
-            act = int(args[0])
-        except ValueError:
-            return '', 'Invalid job: {}\n'.format(args[0])
-        if act not in builtins.__xonsh_all_jobs__:
-            return '', 'Invalid job: {}\n'.format(args[0])
-    else:
-        return '', 'fg expects 0 or 1 arguments, not {}\n'.format(len(args))
-    builtins.__xonsh_active_job__ = act
-    job = builtins.__xonsh_all_jobs__[act]
-    job['bg'] = False
-    job['status'] = 'running'
-    print_one_job(act)
-    os.kill(job['obj'].pid, signal.SIGCONT)
-
-
-def bg(args, stdin=None):
-    """
-    xonsh command: bg
-
-    Resume execution of the currently active job in the background, or, if a
-    single number is given as an argument, resume that job in the background.
-    """
-    _clear_dead_jobs()
-    if len(args) == 0:
-        # start active job in foreground
-        act = builtins.__xonsh_active_job__
-        if act is None:
-            return '', 'Cannot send nonexistent job to background.\n'
-    elif len(args) == 1:
-        try:
-            act = int(args[0])
-        except ValueError:
-            return '', 'Invalid job: {}\n'.format(args[0])
-        if act not in builtins.__xonsh_all_jobs__:
-            return '', 'Invalid job: {}\n'.format(args[0])
-    else:
-        return '', 'bg expects 0 or 1 arguments, not {}\n'.format(len(args))
-    builtins.__xonsh_active_job__ = act
-    job = builtins.__xonsh_all_jobs__[act]
-    job['bg'] = True
-    job['status'] = 'running'
-    print_one_job(act)
-    os.kill(job['obj'].pid, signal.SIGCONT)
+
+import platform
+from warnings import warn
+
+try:

```

Comment:
The idea I was going for was to test for finer-grained features and use them or fall back rather than inferring whether the features are available from a coarse-grained check of platform.system().  I don't think it's a clear win in the this case, because a minor error in jobs_posix.py could cause it to fall back to the non-implemented case and mask other issues (which is why I added a warning).  I'll revisit it when I move jobs to a subpackage, especially since @wrywerytwreywery already has improved the jobs support for Windows.